apiserver/facades/client/application/application.go

```diff
@@ -1777,6 +1796,26 @@ func (api *APIBase) DestroyUnit(args params.DestroyUnitsParams) (params.DestroyU
 		if !unit.IsPrincipal() {
 			return nil, errors.Errorf("unit %q is a subordinate", name)
 		}
+
+		// TODO(wallyworld) - enable-ha is how we remove controllers at the moment
+		// Remove this check before 3.0 when enable-ha is refactored.
+		appName, _ := names.UnitApplication(unitTag.Id())
+		ch, ok := appCharms[appName]
+		if !ok {
+			app, err := api.backend.Application(appName)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			ch, _, err = app.Charm()
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			appCharms[appName] = ch
+		}
+		if ch.Meta().Name == bootstrap.ControllerCharmName {
+			return nil, errors.NotSupportedf("remove units from the controller application")
```

Comment:
Similar: should it read "remov*ing* units ... not supported"?

Reply:
fixed